src/testdir/test_ins_complete.vim

```diff
@@ -2764,24 +2769,26 @@ func Test_complete_fuzzy_match()
   call assert_equal('你好 他好', getline('.'))
 
   " issue #15526
-  set completeopt=fuzzy,menuone,menu,noselect
+  set completeopt=menuone,menu,noselect
   call setline(1, ['Text', 'ToText', ''])
   call cursor(3, 1)
   call feedkeys("STe\<C-X>\<C-N>x\<CR>\<Esc>0", 'tx!')
   call assert_equal('Tex', getline('.'))
 
-  " clean up
-  set omnifunc=
+  " working with longest
+  set completeopt=menu,longest
+  set cfc=k
+  call setline(1, ["iaax", "iaay", "iaaz" , "iay", "ixa", ''])
+  " match iaax iax and i is common longest
+  exe "normal Sx\<C-X>\<C-N>\<esc>"
+  call assert_equal("i", getline('.'))
```

Comment:
longest can also work with completefuzzycollect, but it may use common string to replace the input leader. how do you feel here ? @ychin  @roccomao

Reply:
Thank you very much for your work again, I may test it later and give you some feedback.

Reply:
> `longest` can also work with `completefuzzycollect`

It only supports use with `completefuzzycollect=k` now ? I tested `completefuzzycollect=w` and it seems to have some problems. I didn't test `completefuzzycollect=f` with `longest`.

Reply:
currently yup. mostly want to hear opinions on this behavior now. because the common substring will replace the input leader ..

Reply:
yep.. its strange behavior.

Reply:
Just now, I open a real file, As expect, the leading characters are deleted always. Because more keywords, then fewer common substring.

Reply:
I just submitted an update. Tried the new longest implementation on filename completion. Here's what it looks like. I'll support other modes later if you like them.  minimal vimrc is and files are `touch helio hello`
```
set nocompatible
set cfc=k,f
set completeopt=menu,menuone,longest
```

![test1](https://github.com/user-attachments/assets/ea47f73b-5c14-4bac-98ce-6bb25f57f5c4)

Reply:
I have update fuzzy longest, could you try it when you have available and give me a feedkback.  maybe some edge cases that still need to be dealt with, but let's make sure we are heading in the right direction.

Reply:
Just now, I recompile your latest version, Just doing some very simple testing, it occurred to me that maybe `longest` shouldn't work with `set completefuzzycollect=k`, I'll give you some reasons later. But the current solution looks **OK**, and should be good for people who need it.

Besides, I found a problem in the edge case, when there is only one match, it should be automatically completed. Example is that if there only exist a word `hello`, when enable `longest` and `set completefuzzycollect=k`, then we enter `h` and press `<C-n>`, the `hello` should be auto completed, If the two options are not both set, this behavior is indeed as expected.

Reply:
enable two these two options and `h` auto completed `hello` for me .. something i missed?

Reply:
> enable these two options and h auto completed hello for me .. something i missed?

Sorry, typo, must have called fuzzycollect be actived, You can enter `e` to try, Anyway, not the first character to active `fuzzycollect`

Reply:
i have update for this. should be works now.

Reply:
Why do I say that they are not suitable to work together? Or to be more precise, it is just a little bit regrettable.

Considering the existence of these words: `hello helio think link`, following is the test case with setting `longest` and `set completefuzzycollect=k`. The first column is the input, the second column is the autocomplete, and the third column is the candidate list.

```
h<C-n>  hel - hello
            - helio
            - think

he<C-n> hel - hello
            - helio

ho<C-n> ho  - hello
            - helio

hi<C-n> hi  - think
            - helio

li<C-n> li  - link
            - helio

lo<C-n> lo  - hello
            - helio

in<C-n> in  - link
            - think
```

Everything works as expected and it works great. Unfortunately, if the `longest` option is in **effect**, some content will be automatically completed. If we also set `set completeopt+=fuzzy` at the same time, we may not be able to select some words through the fuzzy filter feature. Because the input is automatically extended now, even if the fuzzy collection is still in the candidate list, we can only select it by pressing keys such as `up`, `down`, or `<C-p>` / `<C-n>`. Corresponding to the example, we cannot select `think` through `h<C-n>k`. But this doesn't always happen, and it shouldn't have any effect in real scenarios.

Even with this small regret, since *fuzzy filter* and *fuzzy collect* are decoupled, this is just a very small problem. You can't have your cake and eat it too. ( 鱼与熊掌不可兼得 )

Reply:
> i have update for this. should be works now.

Yes, everything seems to be working fine at the moment. Thanks again for your work. 👍

Reply:
> Corresponding to the example, we cannot select think through h<C-n>k.

I think you mean `hello think` then `h<C-N>` it auto insert hello you can't use k to fuzzy filter when you also have fuzzy and longest in cot right ? .. this should be a bug if leader is non-consecutive the longest insert should be skip there shouldn't insert hello..

Reply:
I didn't find the bug you mentioned, but it is indeed a bug. 😄

What I mean is that since our input is completed by `longest`, there are fuzzy collected words in the candidate list if set `completefuzzycollect=k`, and we can no longer select words through fuzzy filtering which a good feature. It's a small regret.

Reply:
could you try the new update when you available :) Hope now can works as your expect..

Reply:
> In any case, everything is OK. I am just explaining the irreconcilable issues. As that said: you can't have your cake and eat it too. ( 鱼与熊掌不可兼得 )

yep I'm trying to make it compatible and work in most cases.  no way to support special cases that are difficult to support. The insertion of hello I just mentioned is a bug. about hel then input k want ge think i will try to find a way on it.

Reply:
> The insertion of hello I just mentioned is a bug.

Yes, this is solved now.

> about hel then input k want ge think i will try to find a way on it.

In fact, I think the current solution is OK. Because all the options must cannot exist at the same time, it depends on the trade-offs. I think the current solution is acceptable.

Reply:
> Example is that if there only exist a word hello, when enable longest and set completefuzzycollect=k, then we enter `h` and press `<C-n>`, the hello should be auto completed.

Now this bug comes back 😄

Reply:
fixed and you can get `think` now in this case `hello helio think helpful`

Reply:
Maybe there is a bug, when press `h` then `<C-n>`, complete to `hel`, when enter `k`, now the candidate list is `think`, but we can't select anything, the `up`, `down` and `<C-p>` / `<C-n>` now has no effect.

Your implementation is so timely. Actually, only for myself, I think it is not necessary to implement this.

The original logic is equivalent to `longest` having a higher priority, but now `fuzzy` has a higher priority. I personally think this is inappropriate. Without considering the bug just mentioned, now suppose you enter `h` and then press `<C-n>` to complete to `hel`. At this time, if you press any letter next, such as `x`, `h` + `x` will appear on the screen instead of `hel` + `x` now, this is unacceptable. Secondly, even if `hel` + `x` is displayed on the screen now, I guess your underlying implementation may still only save the input `h`, and then use `h` + `x` to fuzzy filter to achieve the desired effect. In fact, sometimes when you really complete `hel` by `<C-n>`, you may just want to use `hel` + `x` for fuzzy filter.

In short, it's still a trade-off issue. I think `longest` should still be given priority. Since you have chosen longest to help you complete, you must accept the slight loss of other features.

Reply:
> Edit: i think i have a way can make this work input k can also get think. but does this not a very special case ? do we want support this?

Sorry, I didn't see your later edit. I really think the original solution is really good and enough. The new change looks even weirder. Previously, I just gave an extreme example to illustrate the mutual exclusivity between options.

Reply:
I think I know what the problem is. can supported but a bit toublesome. when we can't find a match we can try to fall back and splice the leader for a new match. obviously this makes pum generation slower. But both options work well. nothing is perfect. something has to be sacrificed.